source/extensions/filters/http/transformation/transformation_filter_config.h

```diff
@@ -20,71 +20,85 @@ class Transformation {
       const envoy::api::v2::filter::http::Transformation &transformation);
 };
 
-using TransformationConfigProto = envoy::api::v2::filter::http::RouteTransformations;
+using TransformationConfigProto = envoy::api::v2::filter::http::FilterTransformations;
 using RouteTransformationConfigProto = envoy::api::v2::filter::http::RouteTransformations;
 
 class TransformationFilterConfig : public FilterConfig {
 public:
   TransformationFilterConfig(const TransformationConfigProto &proto_config, const std::string& prefix, Stats::Scope& scope)
-      : FilterConfig(prefix, scope), clear_route_cache_(proto_config.clear_route_cache()) {
-    if (proto_config.has_request_transformation()) {
-      request_transformation_ =
-          Transformation::getTransformer(proto_config.request_transformation());
-    }
-    if (proto_config.has_response_transformation()) {
-      response_transformation_ = Transformation::getTransformer(
-          proto_config.response_transformation());
+      : FilterConfig(prefix, scope) {
+    
+    for (const auto& rule : proto_config.transformations()) {
+      if (!rule.has_match()) {
+        continue;
+      }
+      TransformerConstSharedPtr request_transformation;
+      TransformerConstSharedPtr response_transformation;
+      if (rule.has_route_transformations()) {
+        const auto& route_transformation = rule.route_transformations();
+        clear_route_cache_ = route_transformation.clear_route_cache();
```

Comment:
clear_route_cache_ shouldn't be a member as it will retain the value of the last config.
should probably part of TransformerPair